### PR TITLE
[1/n][dagster-airbyte] Scaffold AirbyteCloudWorkspace and AirbyteClient for rework

### PR DESCRIPTION
**Pull request by maximearmstrong**
## Summary & Motivation

Builds out a very barebones resource and client classes for the new version of the Airbyte cloud integration.

## How I Tested These Changes

Tests will be added in subsequent PRs.

### CODE REVIEW

**Comment by maximearmstrong** [comment]
* **#26205** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26205?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26204** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26204?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26204?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py [hunk 1]
```diff
@@ -791,3 +794,114 @@ def airbyte_cloud_resource(context) -> AirbyteCloudResource:
...
+    @classmethod
+    def from_destination_details(
+        cls,
+        destination_details: Mapping[str, Any],
+    ) -> "AirbyteDestination":
+        raise NotImplementedError()
+
+
+@record
+class AirbyteWorkspaceData:
```
Comment (maximearmstrong) [inline python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py:831]: The workspace data is not specific to Airbyte Cloud, which is why this object is named `AirbyteWorkspaceData`.

File: python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py [hunk 2]
```diff
@@ -791,3 +794,114 @@ def airbyte_cloud_resource(context) -> AirbyteCloudResource:
...
+    """A record representing all content in an Airbyte workspace.
+    Provided as context for the translator so that it can resolve dependencies between content.
+    """
+
+    connections_by_id: Mapping[str, AirbyteConnection]
+    destinations_by_id: Mapping[str, AirbyteDestination]
+
+
+@experimental
+class AirbyteClient:
```
Comment (maximearmstrong) [inline python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py:841]: The workspace data is not specific to Airbyte Cloud - the REST API can be used [for OSS and Cloud instances](https://docs.airbyte.com/api-documentation) - which is why this object is named `AirbyteClient` .

File: python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py [hunk 3]
```diff
@@ -791,3 +794,115 @@ def airbyte_cloud_resource(context) -> AirbyteCloudResource:
 
     """
     return AirbyteCloudResource.from_resource_context(context)
+
+
+# ------------------
+# Reworked resources
+# ------------------
```
Comment (dpeng817) [inline python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py:801]: Maybe v2 to establish clear continuity.

File: python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py (hunk 1 above)
Comment (dpeng817) [reply python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py:831]: feels worth mentioning that in the docstring, that it applies to both airbyte OSS and airbyte cloud potentially? Or at least a comment

**Review by dpeng817 — APPROVED** [review]

**Pull request merged**